recipients_app/lib/data/models/survey/survey.dart

```diff
@@ -0,0 +1,109 @@
+import "dart:convert";
+
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:collection/collection.dart";
+import "package:equatable/equatable.dart";
+
+class Survey extends Equatable {
+  final String id;
+  final SurveyServerStatus? status;
+  final Timestamp? dueDateAt;
+  final Timestamp? completedAt;
+  final String? accessEmail;
+  final String? accessPassword;
+
+  const Survey({
+    required this.id,
+    this.status,
+    this.dueDateAt,
+    this.completedAt,
+    this.accessEmail,
+    this.accessPassword,
+  });
+
+  @override
+  List<Object?> get props {
+    return [
+      id,
+      status,
+      dueDateAt,
+      completedAt,
+      accessEmail,
+      accessPassword,
+    ];
+  }
+
+  Map<String, dynamic> toMap() {
```

Comment:
if we merge the json_serialization branch this can be changed to auto generated methods

Reply:
made use of json_serialization [5789e6a](https://github.com/socialincome-san/public/pull/413/commits/5789e6a81085dfa5f92c82807d37f8388eee8a24)